Cleanup stdlib reals. (#11725)
coq/coq: doc/changelog/10-standard-library/11725-cleanup-reals.rst

- **Changed:**

>   Use implicit arguments for ConstructiveReals. Move ConstructiveReals

Reviewer: ```suggestion
Use implicit arguments for ``ConstructiveReals``. Move ``ConstructiveReals``
```

By "Use implicit arguments", you mean that the `R:ConstructiveReals` argument of lemmas is now implicit, right. Maybe, this can be told more explicitly.

>   into new directory Abstract. Remove imports of implementations inside

Reviewer: ```suggestion
  into new directory ``Abstract``. Remove imports of implementations inside
```

>   those Abstract files. Move Cauchy reals in new directory Cauchy.

Reviewer: ```suggestion
  those ``Abstract`` files. Move implementation by means of Cauchy sequences in new directory ``Cauchy``.
```
